src/Compiler/pars.fsy

```diff
@@ -4430,6 +4430,34 @@ declExpr:
      { let trivia: SynExprYieldOrReturnTrivia = { YieldOrReturnKeyword = rhs parseState 1 }
        SynExpr.YieldOrReturn(($1, not $1), $2, (unionRanges (rhs parseState 1) $2.Range), trivia) }
 
+  | YIELD declExpr COLON typ
+     { let trivia: SynExprYieldOrReturnTrivia = { YieldOrReturnKeyword = rhs parseState 1 }
+       let typedExpr = SynExpr.Typed($2, $4, unionRanges $2.Range $4.Range)
```

Comment:
Could you add language feature check and recovery here as well?
It could reuse the same languagefeature from the previous PR, what is fine

Reply:
Done